planner/core/common_plans.go

```diff
@@ -1537,7 +1538,12 @@ func (e *Explain) getOperatorInfo(p Plan, id string) (string, string, string, st
 	}
 	estCost := "N/A"
 	if pp, ok := p.(PhysicalPlan); ok {
-		estCost = strconv.FormatFloat(pp.Cost(), 'f', 2, 64)
+		if p.SCtx().GetSessionVars().EnableNewCostInterface {
+			planCost, _ := pp.CalPlanCost(property.RootTaskType)
```

Comment:
Why do we need to calculate the cost again here?

Reply:
Actually, we didn't calculate it again here, since our costs are memorized and the cost of the plan must be calculated at the physical optimization stage, so what here does is just using this standard interface `CalPlanCost()` to get the existed cost of this plan.

Reply:
So we are going to remove the `Cost()` method on `PhysicalPlan`, right?

Reply:
YES!

Reply:
Then maybe `GetCost()` is better than `CalPlanCost()`? Because it may not actually "calculate" the cost if it's cached.

Reply:
+1, maybe `CalcCost`/`GetCost` is better.

Reply:
Emm, The key work `Plan` is used to indicate that `CalPlanCost` returns the whole cost of the plan instead of the current operator's like `PhysicalIndexLookUpReader.GetCost`.
And combining the get-method and cal-method together into one seems acceptable since some other plan methods use the same approach like `DeriveStats` and `FindBestTask`; If we split it into two separate methods, I'm afraid it may be hard to use, for example:
```
func (plan) CalPlanCost() (float, error) {
    ...
    childCost, init := plan.child.GetCost()
    if !init {
        childCost, err = plan.child.CalcCost()
        if err != nil {
            return err
        }
    }
    ...
}
```